joblib/test/test_func_inspect.py

```diff
@@ -81,58 +70,69 @@ def f(self, x):
 ###############################################################################
 # Tests
 
-def test_filter_args():
-    yield assert_equal, filter_args(f, [], (1, )), {'x': 1, 'y': 0}
-    yield assert_equal, filter_args(f, ['x'], (1, )), {'y': 0}
-    yield assert_equal, filter_args(f, ['y'], (0, )), {'x': 0}
-    yield assert_equal, filter_args(f, ['y'], (0, ), dict(y=1)), {'x': 0}
-    yield assert_equal, filter_args(f, ['x', 'y'], (0, )), {}
-    yield assert_equal, filter_args(f, [], (0,), dict(y=1)), {'x': 0, 'y': 1}
-    yield assert_equal, filter_args(f, ['y'], (), dict(x=2, y=1)), {'x': 2}
 
-    yield assert_equal, filter_args(i, [], (2, )), {'x': 2}
-    yield assert_equal, filter_args(f2, [], (), dict(x=1)), {'x': 1}
+@parametrize(['func', 'args', 'filtered_args'],
+             [(f, [[], (1, )], {'x': 1, 'y': 0}),
+              (f, [['x'], (1, )], {'y': 0}),
+              (f, [['y'], (0, )], {'x': 0}),
+              (f, [['y'], (0, ), dict(y=1)], {'x': 0}),
+              (f, [['x', 'y'], (0, )], {}),
+              (f, [[], (0,), dict(y=1)], {'x': 0, 'y': 1}),
+              (f, [['y'], (), dict(x=2, y=1)], {'x': 2}),
+              (g, [[], (), dict(x=1)], {'x': 1}),
+              (i, [[], (2, )], {'x': 2})])
+def test_filter_args(func, args, filtered_args):
+    assert filter_args(func, *args) == filtered_args
 
 
 def test_filter_args_method():
     obj = Klass()
     assert filter_args(obj.f, [], (1, )) == {'x': 1, 'self': obj}
 
 
-def test_filter_varargs():
-    yield assert_equal, filter_args(h, [], (1, )), \
-                            {'x': 1, 'y': 0, '*': [], '**': {}}
-    yield assert_equal, filter_args(h, [], (1, 2, 3, 4)), \
-                            {'x': 1, 'y': 2, '*': [3, 4], '**': {}}
-    yield assert_equal, filter_args(h, [], (1, 25), dict(ee=2)), \
-                            {'x': 1, 'y': 25, '*': [], '**': {'ee': 2}}
-    yield assert_equal, filter_args(h, ['*'], (1, 2, 25), dict(ee=2)), \
-                            {'x': 1, 'y': 2, '**': {'ee': 2}}
+@parametrize(['func', 'args', 'filtered_args'],
+             [(h, [[], (1, )],
+               {'x': 1, 'y': 0, '*': [], '**': {}}),
+              (h, [[], (1, 2, 3, 4)],
+               {'x': 1, 'y': 2, '*': [3, 4], '**': {}}),
+              (h, [[], (1, 25), dict(ee=2)],
```

Comment:
For consistency you may as well use {'ee': 2}. To be honest I am not sure why `dict` is used maybe some people find it more explicit.

If you do that do it everywhere in this file.